engine/schema/src/main/java/com/cloud/upgrade/dao/Upgrade41600to41610.java

```diff
@@ -57,6 +60,12 @@ public boolean supportsRollingUpgrade() {
 
     @Override
     public void performDataMigration(Connection conn) {
+        correctGuestOsIdsInHypervisorMapping(conn);
+    }
+
+    private void correctGuestOsIdsInHypervisorMapping(final Connection conn) {
+        LOG.debug("Correcting guest OS ids in hypervisor mappings");
+        guestOsMapper.updateGuestOsIdInHypervisorMapping(conn, 10, "Ubuntu 20.04 LTS", new Ternary<String, String, String>("Xenserver", "8.2.0", "Ubuntu Focal Fossa 20.04"));
```

Comment:
This would fix the incorrect guest os mapping in 4.15.0 to 4.15.1 schema, when upgrading to 4.16.1 from existing 4.16.0 env.

https://github.com/apache/cloudstack/blob/571339f8724fc70c472f55f254125f0c6070d5de/engine/schema/src/main/resources/META-INF/db/schema-41500to41510.sql#L55-L57